frontend/javascripts/oxalis/model/sagas/split_boundary_mesh_saga.ts

```diff
@@ -0,0 +1,112 @@
+
+// The clean up function removes the surface from the scene controller.
+let cleanUpFn: (() => void) | null = null;
+
+// This info object contains data about the tree that is currently
+// adapted to not show its edges (because splines are rendered instead).
+// When the split toolkit is left or when another tree is activated,
+// the original value of edgesAreVisible is restored.
+let temporarilyChangedTreeInfo: {
+  treeId: number;
+  originalEdgesAreVisible: boolean;
+} | null = null;
+
+function* restoreApperanceOfTree() {
+  if (temporarilyChangedTreeInfo == null) {
+    return;
+  }
+  yield* put(
+    setTreeEdgeVisibilityAction(
+      temporarilyChangedTreeInfo.treeId,
+      temporarilyChangedTreeInfo.originalEdgesAreVisible,
+    ),
+  );
+  temporarilyChangedTreeInfo = null;
+}
+
+function* updateSplitBoundaryMesh() {
+  if (cleanUpFn != null) {
+    cleanUpFn();
+    cleanUpFn = null;
+  }
+
+  const isSplitToolkit = yield* select(
+    (state) => state.userConfiguration.activeToolkit === "SPLIT_SEGMENTS",
+  );
+  if (!isSplitToolkit) {
+    yield* call(restoreApperanceOfTree);
+    return;
+  }
+
+  const sceneController = yield* call(() => getSceneController());
+
+  const activeTree = yield* select((state) => getActiveTree(state.annotation.skeleton));
+
+  if (activeTree?.treeId !== temporarilyChangedTreeInfo?.treeId) {
+    // The active tree changed.
+    // Restore the appearance of the old tree.
+    yield* call(restoreApperanceOfTree);
+
+    // Update the appearance of the current tree.
+    if (activeTree != null) {
+      yield* put(setTreeEdgeVisibilityAction(activeTree.treeId, false));
+    }
+
+    // Update temporarilyChangedTreeInfo
+    temporarilyChangedTreeInfo =
+      activeTree != null
+        ? {
+            treeId: activeTree.treeId,
+            originalEdgesAreVisible: activeTree.edgesAreVisible,
+          }
+        : null;
+  }
+
+  if (activeTree?.isVisible) {
+    const nodes = Array.from(activeTree.nodes.values());
+    const points = nodes.map((node) => node.untransformedPosition);
+    if (points.length >= 2) {
+      cleanUpFn = sceneController.addSplitBoundaryMesh(points);
+    }
+  }
+}
+
+export function* splitBoundaryMeshSaga(): Saga<void> {
+  cleanUpFn = null;
+  temporarilyChangedTreeInfo = null;
+  yield* takeWithBatchActionSupport("INITIALIZE_SKELETONTRACING");
+  yield* ensureWkReady();
+
+  // Call once for initial rendering
+  yield* call(updateSplitBoundaryMesh);
+  yield* takeEvery(
+    [
+      "SET_ACTIVE_TREE",
+      "SET_ACTIVE_TREE_BY_NAME",
+      "SET_ACTIVE_NODE",
+      "CREATE_NODE",
+      "DELETE_NODE",
+      "CREATE_TREE",
+      "DELETE_TREE",
+      "SET_TREE_VISIBILITY",
+      "TOGGLE_TREE",
+      "SET_NODE_POSITION",
+      "SET_TRACING",
+      (action: Action) =>
+        action.type === "UPDATE_USER_SETTING" && action.propertyName === "activeToolkit",
+    ] as ActionPattern,
+    updateSplitBoundaryMesh,
+  );
```

Comment:
What about the following actions. IMO they can also manipulate nodes:
- MERGE_TREES
- SET_NODE_POSITION

Reply:
good point. I went through `SkeletonTracingSaveRelevantActions` and added the ones that can be relevant.